src/PrestaShopBundle/Controller/Admin/Sell/Order/OrderController.php

```diff
@@ -709,10 +712,15 @@ public function addProductAction(int $orderId, Request $request): Response
         /** @var OrderForViewing $orderForViewing */
         $orderForViewing = $this->getQueryBus()->handle(new GetOrderForViewing($orderId));
 
-        $updatedProducts = $orderForViewing->getProducts()->getProducts();
+        $updatedProducts = [];
+        foreach ($orderForViewing->getProducts()->getProducts() as $orderProductForViewing) {
+            $updatedProducts[$orderProductForViewing->getOrderDetailId()] = $orderProductForViewing;
+        }
         $newProducts = [];
-        for ($i = count($previousProducts); $i < count($updatedProducts); ++$i) {
-            $newProducts[] = $updatedProducts[$i];
+        foreach (array_keys($updatedProducts) as $productKey) {
+            if (!in_array($productKey, array_keys($previousProducts), true)) {
+                $newProducts[] = $updatedProducts[$productKey];
+            }
         }
```

Comment:
```suggestion
        $newProducts = array_diff(array_keys($updatedProducts), array_keys($previousProducts));
```

Reply:
Thanks Franck. I will only get the keys if I do this. And after that I can put the objects into $newProducts

Reply:
Correct.

```suggestion
        $newProducts = array_diff_assoc($updatedProducts, $previousProducts);
```

Reply:
array_diff_assoc works only if values can be converted to string

I'm trying https://www.php.net/manual/fr/function.array-diff-key.php